com/paypal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package com.paypal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
```

Comment:
if you're creating a singleton make use of it in the below code, isPerfect, getDivisors method, if there is no state to save on this class get rid of this code